### ISSUE TITLE
List-Ops foldr implementation/tests

### ISSUE THREAD

**Issue by hobbitjoe** [exercism/xfsharp#314]
The id and append test cases:
```
//id
foldr (fun item acc -> item :: acc) [] [1 .. big]
//append
foldr (fun item acc -> item :: acc) [100 .. big] [1 .. 99]
```
The arguments for the accumulator and the list to fold over are backwards compared to [F#'s own foldBack function](https://msdn.microsoft.com/en-us/visualfsharpdocs/conceptual/list.foldback%5b't,'state%5d-function-%5bfsharp%5d) which is F#'s equivalent of foldr.

Below are two implentations that have been submitted for foldr
```
let foldr fn lst acc = foldl (flip fn) acc (reverse lst)
let foldr f acc xs = foldl (flip f) acc (reverse xs)
```
The top one I wrote has the function arguments in the order that would be expected of foldBack, while the second one has the arguments in the order of [haskells foldr](http://hackage.haskell.org/package/base-4.9.1.0/docs/Prelude.html#v:foldr).  Was this the intended behavior for this exercise?  Would it make more sense for this exercise to create drop in replacements of the existing functions we are recreating?

**Comment by ErikSchierboom** [comment]
@hobbitjoe Ah, good catch! This is almost certainly due to the fact that I used my Haskell exercise as the basis for the F# one. Would you perhaps like to submit a PR to fix this? If not, no problem!

**Comment by hobbitjoe** [comment]
@ErikSchierboom sure, I can work on a PR.  Should I just swap the lists in the test cases where foldr is used or should the name of the function also be changed?

**Comment by ErikSchierboom** [comment]
Great! I think swapping would be enough, as the exercise's [canonical data](https://github.com/exercism/x-common/blob/master/exercises/list-ops/canonical-data.json) also lists the function as `foldr` (which nicely mirrors the `foldl` function).

**Comment by hobbitjoe** [comment]
PR #316 created for this issue

**Comment by ErikSchierboom** [comment]
Closed due to #316 being merged. Once again thanks @hobbitjoe!
